src/NuGet.Clients/PackageManagement.PowerShellCmdlets/NuGetPowerShellBaseCommand.cs

```diff
@@ -275,24 +277,75 @@ protected override sealed void ProcessRecord()
+        /// </summary>
+        /// <param name="source">The source string specified by -Source switch.</param>
+        /// <returns>Returns an object of PackageSource if the specified source string is a known source. Else returns a null.</returns>
+        protected PackageSource GetMatchingSource(string source)
+        {
+            var packageSources = _sourceRepositoryProvider.PackageSourceProvider?.LoadPackageSources();
+            var matchingSource = packageSources
+                ?.FirstOrDefault(p => StringComparer.OrdinalIgnoreCase.Equals(p.Name, source) ||
+                          StringComparer.OrdinalIgnoreCase.Equals(p.Source, source));
+            return matchingSource;
+        }
+
+        /// <summary>
+        /// Checks if the sourse is valid http, local or known source. Else throws an exception.
         /// </summary>
         /// <param name="source">The source string specified by -Source switch.</param>
-        protected void UpdateActiveSourceRepository(string source)
+        /// <param name="packageId">The package ID string specified.</param>
+        /// <param name="matchingSource">Matching Source if the specified source is a known source. Else null.</param>
+        /// <returns>Returns an absolute version of the source string if it is a relative local source. Else returns it unchanged.</returns>
+        /// <exception cref="InvalidOperationException">Throws an InvalidOperationException if the specified source is either invalid or is not http/local type.</exception>
+        protected string CheckSourceValidity(string source)
         {
-            var packageSources = _sourceRepositoryProvider?.PackageSourceProvider?.LoadPackageSources();
+            // Check validity only if the source and package Id were specified and it is not a known source
+            if (!string.IsNullOrEmpty(source))
+            {
+                // Convert a relative URI into an absolute URI
+                source = ConvertRelativeUriToAbsolute(source);
+
+                // Convert source into a PackageSource
+                var packageSource = new PackageSource(source);
+
+                // Check if the source is a valid http or local source
+                if ((packageSource.IsHttp && packageSource.TrySourceAsUri == null) || (packageSource.IsLocal && !Directory.Exists(packageSource.Source)))
+                {
+                    var invalidOperationException = new InvalidOperationException();
+                    invalidOperationException.Data[Strings.ExceptionType] = Strings.UnknownSource;
+                    throw invalidOperationException;
+                }
+                // If there was no matching known source
+                else if (!packageSource.IsHttp && !packageSource.IsLocal)
+                {
+                    var invalidOperationException = new InvalidOperationException();
+                    invalidOperationException.Data[Strings.ExceptionType] = Strings.UnknownSourceType;
```

Comment:
I like the idea of throwing a custom exception. Let's go a bit further and introduce new exception type for that.
Some good examples:
NuGet.Client/src/NuGet.Clients/NuGet.CommandLine/Common/CommandLineException.cs 
NuGet.Client/src/NuGet.Core/NuGet.Configuration/Exceptions/NuGetConfigurationException.cs